x/emissions/module/rewards_calc.go

```diff
@@ -125,6 +130,54 @@ func GetParticipantEmissionsForTopic(
 	return rewards, nil
 }
 
+// This function checks topic weights and then masks them if not enough inferences were collected in that timestep
+// It should:
+//
+//	mask the inputted weights above with GetNumInferencesInRewardEpoch(),
+//	checking inference cadence by reward epoch length / topic inference cadence
+//	simple forgiveness check (only if many inferences are missing, don't reward)
+func MaskWeightsIfInsufficientLiveness(
```

Comment:
I would sleep much better at night if we had a unit test for this function that actually ensures that the function does as we expect

Reply:
Also this function probably does not need to be exported out of this module and can be lowercased